src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1011,6 +1019,11 @@ internal ValueTask<IBackendMessage> ReadMessage(bool async, DataRowLoadingMode d
                 return ReadMessageLong(dataRowLoadingMode, readingNotifications2: false);
             }
 
+            // Just in case if we were successful in sending a cancellation, but the query is already completed (and the response is already in the buffer)
+            // Same thing is done below for the cases, when RFQ is not in the buffer
+            if (messageCode == BackendMessageCode.ReadyForQuery)
```

Comment:
Let's move this into the switch above, it should make the code slightly more efficient.